### PR TITLE
Review CRUD!! 

### PR DESCRIPTION
**Pull request by luisa-li**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/69)
Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

See screenshots 

Screenshot 2026-02-08 at 6 06 41 PM
Screenshot 2026-02-08 at 6 05 06 PM
Screenshot 2026-02-08 at 6 04 29 PM
Screenshot 2026-02-08 at 6 03 52 PM

# How Has This Been Tested?
Please describe the tests that you manually ran to verify your changes (beyond any unit/integration tests written and ran).

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]

### Android
[Add Android screenshots here]

### CODE REVIEW

File: backend/internal/models/review.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	"github.com/google/uuid"
+)
+
+type Review struct {
+	ID             uuid.UUID `json:"id" db:"id" doc:"Unique review identifier"`
+	RegistrationID uuid.UUID `json:"registration_id" db:"registration_id" doc:"ID of the linked registration"`
+	GuardianID     uuid.UUID `json:"guardian_id" db:"guardian_id" doc:"ID of the guardian"`
+	Description    string    `json:"description" db:"description" doc:"The review text"`
+	Categories     []string  `json:"categories" db:"categories" doc:"Review categories for this review, can be one of fun, engaging, interesting or informative."`
```
Comment (josh-torre) [inline backend/internal/models/review.go]: We should also create a type + validate the category input. See the attached docs and example code

### Example

```go
type ReviewCategory string

// When you change the below type you must also update the list ReviewCategories
const (
    Fun   ReviewCategory = "fun"
    Engaging  ReviewCategory = "engaging"
    Interesting ReviewCategory = "interesting"
    Informative ReviewCategory = "informative"
)

// When you change the below list you must also update the type ReviewCategory
var ReviewCategories = []ReviewCategory{
    Fun,
    Engaging,
    Interesting,
    Informative,
}

func (Status) Schema(r huma.Registry) *huma.Schema {
    return &huma.Schema{
        Type: "string",
        Enum: ReviewCategories,
    }
}

type UpdateOrderRequest struct {
    Body struct {
        Status Status `json:"ReviewCategory" doc:"Categories the user used to describe their experience"`
    }
}
```



### Docs
[https://huma.rocks/features/schema-customization/?#field-schema](https://huma.rocks/features/schema-customization/?#field-schema)

File: backend/internal/models/review.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	CreatedAt      time.Time `json:"created_at" db:"created_at" doc:"Timestamp when registration was created"`
+	UpdatedAt      time.Time `json:"updated_at" db:"updated_at" doc:"Timestamp when registration was last updated"`
+}
+
+type CreateReviewInput struct {
+	Body struct {
+		RegistrationID uuid.UUID `json:"registration_id" db:"registration_id" doc:"ID of the linked registration"`
+		GuardianID     uuid.UUID `json:"guardian_id" db:"guardian_id" doc:"ID of the guardian"`
+		Description    string    `json:"description" db:"description" doc:"The review text"`
+		Categories     []string  `json:"categories" db:"categories" doc:"Review categories for this review, can be one of fun, engaging, interesting or informative."`
```
Comment (josh-torre) [inline backend/internal/models/review.go]: Same thing here, we should be validating the Categories

File: frontend/bun.lock [hunk 3]
```diff
@@ -0,0 +1,2661 @@
+{
```
Comment (josh-torre) [inline frontend/bun.lock]: There shouldnt be any changes to the bun.lock in this PR
